src/WebJobs.Script.WebHost/SecretManager.cs

```diff
@@ -109,41 +109,53 @@ public virtual FunctionSecrets GetFunctionSecrets(string functionName)
-            // It is important that we determine functions based on the presence of a folder,
-            // not whether we've identified a valid function from that folder. This ensures
-            // that we don't delete logs/secrets for functions that transition into/out of
-            // invalid unparsable states.
-            var functionLookup = Directory.EnumerateDirectories(rootScriptPath).ToLookup(p => Path.GetFileName(p), StringComparer.OrdinalIgnoreCase);
+                // Create a lookup of all potential functions (whether they're valid or not)
+                // It is important that we determine functions based on the presence of a folder,
+                // not whether we've identified a valid function from that folder. This ensures
+                // that we don't delete logs/secrets for functions that transition into/out of
+                // invalid unparsable states.
+                var functionLookup = Directory.EnumerateDirectories(rootScriptPath).ToLookup(p => Path.GetFileName(p), StringComparer.OrdinalIgnoreCase);
 
-            var secretsDirectory = new DirectoryInfo(_secretsPath);
-            foreach (var secretFile in secretsDirectory.GetFiles("*.json"))
-            {
-                if (string.Compare(secretFile.Name, ScriptConstants.HostMetadataFileName, StringComparison.OrdinalIgnoreCase) == 0)
+                var secretsDirectory = new DirectoryInfo(_secretsPath);
+                if (!Directory.Exists(_secretsPath))
                 {
-                    // the secrets directory contains the host secrets file in addition
-                    // to function secret files
-                    continue;
+                    return;
                 }
 
-                string fileName = Path.GetFileNameWithoutExtension(secretFile.Name);
-                if (!functionLookup.Contains(fileName))
+                foreach (var secretFile in secretsDirectory.GetFiles("*.json"))
                 {
-                    try
+                    if (string.Compare(secretFile.Name, ScriptConstants.HostMetadataFileName, StringComparison.OrdinalIgnoreCase) == 0)
                     {
-                        secretFile.Delete();
+                        // the secrets directory contains the host secrets file in addition
+                        // to function secret files
+                        continue;
                     }
-                    catch
+
+                    string fileName = Path.GetFileNameWithoutExtension(secretFile.Name);
+                    if (!functionLookup.Contains(fileName))
                     {
-                        // Purge is best effort
+                        try
+                        {
+                            secretFile.Delete();
+                        }
+                        catch
+                        {
+                            // Purge is best effort
+                        }
                     }
                 }
             }
+            catch
```

Comment:
The issue is a little worse than that - it actually causes a startup time failure that will cause the host to continually recycle, until a function is added.

I could add back some error output in the catch though, but we don't want to block the host from running successfully if we can't purge.